### PR TITLE
Fixes #221 Showing reg exp for detection and FROM intruction.

### LINKED ISSUE #221: More descriptive message in test_from_directive_is_valid
It took me too much time to figure out what is the reason of this test failure:
```
TEST FAIL:  /usr/lib/python2.7/site-packages/moduleframework/tests/static/dockerfile_lint.py:DockerInstructionsTests.test_from_directive_is_valid
     reason -> FROM instruction is not valid.
        More info: https://docs.docker.com/engine/reference/builder/#from
     /home/zdravomil/avocado/job-results/job-2018-03-07T05.38-ae3c4aa/test-results/02-_usr_lib_python2.7_site-packages_moduleframework_tests_static_dockerfile_lint.py:DockerInstructionsTests.test_from_directive_is_valid/debug.log
```
with test data:
```
FROM fedora
...
```

The message should describe what is wrong with the base image name (missing registry, ...)

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

This Pull request adds more information, why FROM instruction is not valid.

New output looks like:
```
TEST FAIL:  ../../moduleframework/tests/static/dockerfile_lint.py:DockerInstructionsTests.test_from_directive_is_valid
     reason -> FROM instruction (baseruntime-*/+54/baseruntime:latest) is not valid. We use reg ^[a-z0-9.]+(\/[a-z0-9\D.]+)+$.
	More info: https://docs.docker.com/engine/reference/builder/#from
     /home/phracek/avocado/job-results/job-2018-03-12T17.25-6f28f24/test-results/02-.._.._moduleframework_tests_static_dockerfile_lint.py_DockerInstructionsTests.test_from_directive_is_valid/debug.log
```

### CODE REVIEW

**Comment by phracek** [comment]
Yes, we have to update regexp.

File: moduleframework/dockerlinter.py [hunk 1]
```diff
@@ -108,7 +108,22 @@ def _get_structure_as_dict(self):
...
+    def get_from_instruction(self):
+        """
+        Function returns FROM instruction from Dockerfile
+        :return: string or None
+        """
+        struct = self.dfp_structure[0]
+        if struct.get(INSTRUCT) == 'FROM':
+            return struct.get('value')
+        else:
+            return None
```
Comment (jscotka) [inline moduleframework/dockerlinter.py]: None is default return value of function, so ``else`` can be removed

**Comment by dhodovsk** [comment]
I have some difficulties running the linter tests on localhost:
```
[dhodovsk@localhost meta-test-family]$ git status 
HEAD detached at phracek/issue_221
nothing to commit, working tree clean
[dhodovsk@localhost meta-test-family]$ make check-linter 
cd examples/linter/tools && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/tools/mtf -l
/bin/sh: /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/tools/mtf: No such file or directory
make: *** [Makefile:12: check-linter] Error 127
[dhodovsk@localhost meta-test-family]$ 

```

Is there something that needs to be done before executing `make check-linters`?

**Comment by phracek** [comment]
`make check-linter` is something else. Try `make -C examples/mtf-linters check-linters`.
It seems like `Makefile's` in MTF have to be reviewed.

**Comment by phracek** [comment]
Nevertheless, I have fixed the issue.

**Comment by dhodovsk** [comment]
It seems that linter tests are still not run.
This is the output of make check-linter:
```
[dhodovsk@localhost meta-test-family]$ make check-linter 
cd examples/linter/tools && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/moduleframework/mtf_scheduler.py -l
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
cd examples/linter/rhscl-postgresql && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/moduleframework/mtf_scheduler.py -l
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
cd examples/linter/rhscl-nginx && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/moduleframework/mtf_scheduler.py -l
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
cd examples/linter/f26-etcd && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/moduleframework/mtf_scheduler.py -l
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
cd examples/linter/f26-flannel && PYTHONPATH=/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family MODULE=docker /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/moduleframework/mtf_scheduler.py -l
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
MTF config dir exists, search for /usr/share/moduleframework/mtf.conf.d/*.yaml
MTF config load: /usr/share/moduleframework/mtf.conf.d/mtf.yaml
```

Also, there is a failure when I run linter tests with mentioned command:
```
[dhodovsk@localhost meta-test-family]$ make -C examples/mtf-linters check-linters
make: Entering directory '/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/examples/mtf-linters'
pytest test_linters.py
========================================== test session starts ==========================================
platform linux2 -- Python 2.7.14, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family, inifile:
plugins: catchlog-1.2.2
collected 5 items                                                                                        

test_linters.py ..F..

=============================================== FAILURES ================================================
_________________________________________ test_docker_from_dash _________________________________________

    def test_docker_from_dash():
        Dockerfile_FROM = """FROM rhel7:7.5-175"""
        dfl = prepare_linter(Dockerfile_FROM)
>       assert dfl.check_from_directive_is_valid()
E       assert False
E        +  where False = <bound method DockerfileLinter.check_from_directive_is_valid of <moduleframework.dockerlinter.DockerfileLinter object at 0x7f5ea1dae810>>()
E        +    where <bound method DockerfileLinter.check_from_directive_is_valid of <moduleframework.dockerlinter.DockerfileLinter object at 0x7f5ea1dae810>> = <moduleframework.dockerlinter.DockerfileLinter object at 0x7f5ea1dae810>.check_from_directive_is_valid

test_linters.py:52: AssertionError
------------------------------------------- Captured log call -------------------------------------------
parser.py                  351 DEBUG    instructions: {}
================================== 1 failed, 4 passed in 0.19 seconds ===================================
make: *** [Makefile:4: check-linters] Error 1
make: Leaving directory '/home/dhodovsk/repos/github.com/fedora-modularity/meta-test-family/examples/mtf-linters'
```

**Comment by phracek** [comment]
I have tested with `command line instructions` mentioned in this Pull Request and all works fine and properly.

**Comment by dhodovsk** [comment]
I've tried it in clean environment and it runs OK, thank you!

**Pull request merged**